genkan/auth/oAuth.js

```diff
@@ -0,0 +1,182 @@
+// Load environment
+const config = require('../config')
+// Name of theme used in configuration
+const theme = `genkan-theme-${config.genkan.theme}`
+
+// MongoDB
+const MongoClient = require('mongodb').MongoClient
+const url = config.mongo.url
+const dbName = config.mongo.database
+require('../db')
+
+// UUID & Hashing
+const sha512 = require('hash-anything').sha512
+const bcrypt = require('bcrypt');
+const saltRounds = 10;
+
+// Random Password Generator
+const generator = require('generate-password');
+
+// Token Generator
+const tokenGenerator = require('./tokenGenerator')
+
+// NOTE: Will clean up and organise the code here
+
+MongoClient.connect(url, {useUnifiedTopology: true}, function(err, client) {
+    if (err) throw err
+
+    const db = client.db(dbName)
+
+    loginAccountGoogle = (email, googleID, callback) => {
```

Comment:
If this function is a duplicate of that in login.js, it would be better to import login.js and reuse the function instead of repeating the code. Makes it easier to maintain in a situation where we need to change the login mechanic in future.

Reply:
Regarding about that, I created another file because simply I felt its much easier to do it on a blank slate.

Reply:
Hmm... I'll see what I can do to reuse functions.